### ISSUE TITLE
HGVSParseError for gene symbols with "-" (dashes) in them

### ISSUE THREAD

**Issue by davmlaw** [biocommons/hgvs#670]
In theory, gene names shouldn't have dashes in them, but in the wild they do, and "ADAMTSL4-AS1" is an official [HGNC:32041 symbol](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:32041)

```
from hgvs.parser import Parser
parser = Parser()
parser.parse_hgvs_variant("NR_104133.1(ADAMTSL4-AS1):n.101del")
```

gives:

```
ParseError: 
NR_104133.1(ADAMTSL4-AS1):n.101del
                    ^
Parse error at line 1, column 20: expected one of ')', or a letter or digit. trail: [letterOrDigit]
```

while removing the dash, eg

```
parser.parse_hgvs_variant("NR_104133.1(ADAMTSL4):n.101del")
```

works fine

**Comment by davmlaw** [comment]
Looks like we may need "_" (underscore) as well, here are some other approved HGNC symbols that currently fail:

[GTF2H2C_2](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:44196)
[APOBEC3A_B](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:44196)
[C4B_2](https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:42398)

**Comment by davmlaw** [comment]
Much like the previous gene_symbol match disallows numbers at the start of a gene name, I wanted to disallow dashes/underscores at the start and end. I tried to fix this using the obvious regex style:

```
gene_symbol = <letter ('_' | '-' | letterOrDigit)* letterOrDigit+>
```

However I believe the Ometa parser has no backtracking on success (unlike regex which will attempt re-matches if it fails at a higher level).

So - instead I had to allow the dashes/underscore at the end, but then used Python code to validate it

I have done this in a branch off a branch of #661 (store generated Parser code)

https://github.com/davmlaw/hgvs/commit/c5bdd9bb379f22600705b6541bbba6384f083bbd

So I won't do this pull request until that one is merged.

If you reject that one, and want this one against biocommons HGVS master (ie no stored generated Parser Python code) then let me know
